ante/fee.go

```diff
@@ -54,16 +94,53 @@ func (mfd BypassMinFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 				fee := gp.Amount.Mul(glDec)
 				requiredFees[i] = sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt())
 			}
-
-			if !feeCoins.IsAnyGTE(requiredFees) {
+			// 1stake is not a subset of 0stake or 0photon or 1photon, not skip the min gas price check. this case can still use the DenomsSubsetOf() from sdk
+			// empty set is subset of empty set for sdk DenomsSubsetOf
+			// 0 stake is a subset of a 1stake, but the custom IsAnyGTEIncludingZero will be true as well.
+			if !feeCoins.DenomsSubsetOf(requiredFees.Sort()) {
+				return next(ctx, tx, simulate)
+			}
+			// requiredFees here is ensured not all zero, when check min_gas_price, fee might be zero. if min_gas_price=0stake,1photon, and feecoins is 0stake, it should not return err. so use IsAnyGTEIncludingZero() rather than IsAnyGTEIncludingZero() from sdk.
+			if !IsAnyGTEIncludingZero(feeCoins, requiredFees) {
 				return ctx, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFee, "insufficient fees; got: %s required: %s", feeCoins, requiredFees)
 			}
 		}
 	}
 
+	// when the tx is bypass msg type, still need to check the denom is not random denom
```

Comment:
I wouldn't use the nomenclature "random" here, but rather denoms that are not supported for x/globalfee.